stan/math/prim/fun/any.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef STAN_MATH_PRIM_FUN_ANY_HPP
+#define STAN_MATH_PRIM_FUN_ANY_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/functor/for_each.hpp>
+#include <algorithm>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return true if any values in the input are true.
+ *
+ * Overload for a single boolean input
+ *
+ * @param x boolean input
+ * @return The input unchanged
+ */
+inline bool any(bool x) { return x; }
+
+/**
+ * Return true if any values in the input are true.
+ *
+ * Overload for Eigen types
+ *
+ * @tparam Eigen type of the input
```

Comment:
```suggestion
 * @tparam ContainerT A type derived from `Eigen::EigenBase` that has an `integral` scalar type
```

Double check the docs for all these